VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -62,15 +65,26 @@ void ViewItem::insertColumn(int column)
 
 void ViewItem::insertNode(Model::Node* node, int column, int row)
 {
+	auto ref = new Model::NodeReference();
+	ref->setTarget(node);
 	//First, make sure the current grid is big enough to fit the node
 	ensureColumnExists(column);
 	if (nodes_[column].size() <= row)
 		nodes_[column].resize(row + 1);
+	//Find the first empty node after the insertion, return its index
+	auto empty = emptyAfter(column, row);
+	//If it is an empty node -> resize the empty node (if there is a node after)
```

Comment:
You should not resize the empty node here. There are other situations where it needs to be resized (e.g. if some of the other nodes grow/shrink). The empty node (or its equivalent in the new implementaion) should have some knowledge of what items determine its size. This should be used by the visualization of the node to recompute its size which should happen in the determineChildren() method, that you will have to override (and make sure to call Super::determineChildren()).